modules/metadata-support/src/main/java/org/mule/runtime/metadata/api/cache/ConfigurationMetadataCacheIdGenerator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.metadata.api.cache;
+
+import org.mule.runtime.ast.api.ComponentAst;
+
+import java.util.Optional;
+
+/**
+ * Allows to retrieve the {@link MetadataCacheId} for Configs. It's used since the
+ * {@link org.mule.runtime.extension.api.component.ComponentParameterization} has no information of the configs, including its
+ * child elements.
+ *
+ * @since 4.5
+ */
+public interface ConfigurationMetadataCacheIdGenerator {
+
+  /**
+   * Returns the {@link MetadataCacheId} corresponding to the configuration.
+   * 
+   * @param configName      - the configuration name to get the {@link MetadataCacheId}
```

Comment:
why the `-` at the beginning?

Reply:
I based on other similar javadocs, for me it's the same 🤷‍♂️